visualization/app/codeCharta/util/customViewBuilder.ts

```diff
@@ -0,0 +1,140 @@
+"use strict"
+import {State} from "../codeCharta.model"
+import {CustomView} from "../model/customView/customView.api.model";
+import {CustomViewFileStateConnector} from "../ui/customViews/customViewFileStateConnector";
+
+export class CustomViewBuilder {
+	private static readonly CC_CUSTOM_VIEW_API_VERSION = "1.0.0"
+
+	static buildFromState(viewName: string, state: State): CustomView {
+		const customViewFileStateConnector = new CustomViewFileStateConnector(state.files)
+
+		const customView = {
+			name: viewName,
+			mapSelectionMode: customViewFileStateConnector.getMapSelectionMode(),
+			assignedMap: customViewFileStateConnector.getJointMapName(),
+			mapChecksum: customViewFileStateConnector.getChecksumOfAssignedMaps(),
+			customViewVersion: this.CC_CUSTOM_VIEW_API_VERSION
+		} as CustomView
+
+		customView.stateSettings = {
+			appSettings: undefined,
+			dynamicSettings: undefined,
+			fileSettings: undefined,
+			treeMap: undefined
+		}
+
+		this.initializeAppSettings(customView)
+		this.initializeDynamicSettings(customView)
+		this.initializeFileSettings(customView)
+		this.initializeTreeMapSettings(customView)
+
+		this.deepMapOneToOther(state, customView.stateSettings)
+
+		return customView
+	}
+
+	private static initializeAppSettings(target: CustomView) {
```

Comment:
Shouldn't we set it all to the default settings in codeCharta.model, just for consistency?

Reply:
Ok that makes sense, closing the comment